### ISSUE TITLE
BLEURT: Learning Robust Metrics for Text Generation

### ISSUE THREAD

**Issue by toriving** [toriving/Plz_Read_The_Paper#132]
https://arxiv.org/pdf/2004.04696.pdf

![image](https://user-images.githubusercontent.com/42434734/112867162-8bb1ba80-90f5-11eb-8194-9b47f966ca96.png)

![image](https://user-images.githubusercontent.com/42434734/112867178-90766e80-90f5-11eb-9914-bb02ca92248f.png)

**Comment by toriving** [comment]
https://researcher-hj.tistory.com/6

 
![image](https://user-images.githubusercontent.com/42434734/112867662-26aa9480-90f6-11eb-858f-08f8c0cf051b.png)


① BERT pre-training : 그냥 인풋들에 대해 개념적인 단어로 표현하기 위해 쓰인다. (기존 BERT를 가져다 쓴다고 하니 여기에 대해 더 공부해야 할 듯)

② Pre-training on synthetic sentence pairs : 이 부분이 새로 만든 방법으로, 학습할 때 데이터와 테스트하는 데이터의 도메인이나 평가 점수의 편차가 크지 않도록 한다. 이를 위해 위키피디아에서 여러 문장들을 가져와서 단어를 사이사이에 넣거나 지우는 등의 문장의 변형을 한다. 그리고 변형된 문장과 원래의 문장 사이의 여러 기계적 평가 지표들(BLEU)을 계산한다. (구글 AI 블로그 참고)

③ Fine-tuning on public human ratings : 그리고 fine tuning 단계시에는 WMT Metrics 데이터셋과 같은 사람(사용자)이 평가한 지표를 이용한다.

④ (optional) Fine-tuning on application-specific human ratings : 그리고 어떤 애플리케이션을 타겟팅한다면 그와 관련한 데이터셋을 사용하여 fine-tuning을 더 하면 된다.

**Comment by toriving** [comment]
![image](https://user-images.githubusercontent.com/42434734/113531549-0e5ddc80-9604-11eb-861b-b74ef3ff21cf.png)

**Comment by toriving** [comment]
3. Fine-tuning BERT for Quality Evaluation
- BERT(x, x^hat) 을 통해 cls token 의 representation vector를 뽑고 linear layer를 통해 rating을 predict 함 
- Regression 이므로 L2 loss 사용
- 그러나 이 방식은 많은 IID 데이터가 필요하고, 이는 다양한 작업 및 모델로 일반화해야하는 것에 적합하지 않음

4. Pre-Training on Synthetic Data
- key aspect는 Fine tuning전 워밍업 하는데 사용되는 pre-training 기술
- large number of synsthetic reference-candidate pairs (z, z^hat) 을 만들어 several lexical- and semantic level supervision을 사용해 multi task loss로 학습
- 해당 단계를 통해 일반화가 잘됨
- BLEURT가 광범위한 NLG 도메인 및 작업에 대처할 수 있도록 참조 문장 세트는 크고 다양해야함
- 문장 쌍은 다양한 어휘, 구문 및 의미 적 차이를 포함해야함

4.1 Generating Sentence Pairs
- 위키피디아 segment z로 부터 Mask-filling, back-translation, randomly dropping out words를 통해 z^hat을 만든다

4.2 Pre-Traininig Signals
- pre-training signal은 여러개 존재하며 각 task 별 T_k로 표현함
- 좋은 pre-training signal은 다양한 어휘 및 의미 적 차이를 포착해야함. 
- 또한 접근 방법이 대량의 합성 데이터로 확장 될 수 있도록 비용이 저렴해야함.
    - Automatic Metrics : BLEU, ROUGE, BERTScore 사용
    - Backtranslation Likelihood : roundtrip loss (cycling loss) 와 비슷함
    - Textual entailment : entailment loss
    - Backtranslation flag

4.3 Modeling
- 위 loss들을 잘 조합해서 학습함

**Comment by toriving** [comment]
Metric

WMT Metrics Shared Task의 2017-2019년, to-English 언어 쌍 데이터를 사용하였다. 각 연도에 대해, 공식 WMT test set(사람의 평가와 같이 뉴스 도메인에서 얻은 수천 개의 문장 쌍으로 구성)을 사용하였다.
training set은 5360, 9482, 147691개의 예시가 있다. 2018년과 2019년의 test set은 더 noise가 많은데, 좀 더 낮은 상관관계를 보인다.

자동화된 평가 방법과 인간 평가와 일치하는 정도를 평가한다. 각 연도에 대해, Kendall’s Tau 
τ
(실험 간 일관성을 위해)와 공식 WMT 평가(완전성을 위해)를 둘 다 사용하였다. 공식 WMT 평가방법은 Pearson’s correlation이나 DARR라 불리는 Kendall’s Tau의 강건한 변형이며, 부록에서 볼 수 있다. 모든 숫자는 벤치마크의 구현에서 가져왔다. 결과는 전체적으로 공식 결과와 일관되지만 2018년과 2019년에서 약간의 차이를 발견하였고, 표에서 확인할 수 있다.

(https://greeksharifa.github.io/machine_learning/2021/01/13/BLEURT-Learning-Robust-Metrics-for-Text-Generation/#51-wmt-metrics-shared-task)

**Comment by toriving** [comment]
https://github.com/toriving/Plz_Read_The_Paper/blob/master/PAPERS/BLEURT-%20Learning%20Robust%20Metrics%20for%20Text%20Generation.pdf
